### ISSUE TITLE
Support exclusion of files/directories

### ISSUE THREAD

**Issue by michaelmoussa** [adrienverge/yamllint#5]
There doesn't seem to be a way to lint all files in a directory, but exclude some, for example:

```
$
└── foo
    ├── a.yml
    ├── b1.yml
    ├── b2.yml
    ├── b3.yml
    ├── c.yml
    ├── d.yml
    └── e.yml
```

Let's say that, for whatever reason, I don't care about the formatting of some of these yml files. I'd like to be able to run something along the lines of:

```
yamllint foo --exclude=c.yml
```

or

```
yamllint foo --exclude=b*.yml
```

or perhaps multiple...

```
yamllint foo --exclude=b*.yml,d.yml
```

My specific use case is a directory with a bunch of organized yml files that also includes a git submodule with some yml files that would fail linting. It's third-party code and doesn't live in our repository, so I don't need to validate it.

Alternatively, this could be configurable:

```
# .yamllint
files_or_dirs:
    include:
      - foo
      - bar
      - baz.yml
    exclude:
      - foo/b*.yml
      - bar/boo.yml
```

Thanks for sharing this library! It was very easy to get everything going.

**Comment by adrienverge** [comment]
Hi @michaelmoussa, thanks for this idea!

What you describe looks very useful. However I'm not sure such filtering should be part of yamllint's logic, especially because it would be redundant with common UNIX tools.

For instance, what about using `find`?

```bash
yamllint $(find foo -name '*.yml' ! -name 'b*.yml' ! -name 'd.yml')
```

**Comment by michaelmoussa** [comment]
Yes, that's what I've been using for now, and it works fine.

I'm used to having this feature in other linting / sniffing tools so I would have thought I'd see it here too. (examples: [phpcs](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-files-and-folders), [Rubocop](https://github.com/bbatsov/rubocop#includingexcluding-files)

You are correct though that it is redundant with the existing OS functionality - so I totally understand if it's not something you would want to add. Just throwing the suggestion out there. :)

**Comment by quba42** [comment]
This is a standard linter feature for most linters for good reason. Think of CI pipelines that should by default lint all YAML files in the project using `yamllint -s .`, but now I need to exclude some folder that contains auto generated build artifacts or something. There really isn't a good way of doing that with "standard UNIX tools".

**Comment by adrienverge** [comment]
@quba42 this issue is 6 year old. Please check out *Ignoring paths* from [the documentation](https://yamllint.readthedocs.io/en/stable/configuration.html#ignoring-paths).

**Comment by quba42** [comment]
Thanks, that looks great! I wonder what I was doing wrong in my googling to land on six year old issues rather than in the documentation I was looking for! :thinking:
